examples/private_connection/main.tf

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  version = "~> 3.7.0"
+  project = var.project_id
+}
+
+locals {
+  service_list = [
+    "compute.googleapis.com",
+    "cloudresourcemanager.googleapis.com",
+    "serviceusage.googleapis.com",
+    "servicenetworking.googleapis.com",
+    "dns.googleapis.com"
+  ]
+}
+
+resource "google_project_service" "main" {
+  count              = length(local.service_list)
+  project            = var.project_id
+  service            = local.service_list[count.index]
+  disable_on_destroy = "false"
+}
+
+resource "google_compute_network" "network" {
+  name                    = "forseti-network"
+  auto_create_subnetworks = false
+}
+
+resource "google_compute_subnetwork" "subnet" {
+  name                     = "forseti-subnet"
+  region                   = var.region
+  ip_cidr_range            = "10.127.0.0/20"
+  network                  = google_compute_network.network.self_link
+  private_ip_google_access = true
+  log_config {
+    aggregation_interval = "INTERVAL_10_MIN"
+    flow_sampling        = 1
+    metadata             = "INCLUDE_ALL_METADATA"
+  }
+}
+
+# Router to control egress only to Google APIs
```

Comment:
This comment is a little misleading. This router + the nat allows egress to the internet. They are needed only during Forseti installation phase to talk to the public internet. You can use a `count` based on `post_install` to configure provision or not this router and the nat below. They should only be provisioned during installation, and removed post installation.

Reply:
how it can be implemented with a terraform declarative syntax? if it is needed only to access forseti binaries, it is possible to leverage GCS bucket and utilise the same private access as we do for Cloud SQL. WDYT?

Reply:
Example: https://github.com/terraform-google-modules/terraform-google-lb-http/blob/master/main.tf#L25

Reply:
i do not see a value to add additional variable that will let a user to control whether to provision the nat or not since there is no clear indication for that.
i do not work with TF too much. for me running the same plan twice and manually define next time a specific var to 0 just to uninstall stuff is a bit artificiant and burlesque.
however, let me know if it makes sense to you and the usage is clear to the reader

Reply:
So the purpose of this example with the DNS record and route and additional firewalls is after Forseti is installed, you can do two things:
1. Restrict `forseti-server-allow-internet` firewall rule to only allow destination to google VIP
2. remove the nat

Using a variable to togger that is better than manually change the ip destination and remove/comment out the nat blocks.

Reply:
okey, i will use the variable to trigger between provisioning of router/nat and setting destination to _any_ for `"forseti-server-allow-internet"` firewall resource and unprovisioning and setting destination to 199.36.153.8/30.
is it what you have in mind?

Reply:
Yes. Thanks!